internal/template-validator/tlsinfo/tlsinfo.go

```diff
@@ -45,100 +49,73 @@ func (ti *TLSInfo) initCryptoConfig() {
 	ti.sspTLSOptions.MinTLSVersion, _ = os.LookupEnv(TLSMinVersionEnvName)
 }
 
-func (ti *TLSInfo) initCerts() {
+func (ti *TLSInfo) initCerts() error {
+	if ti.stopWatch != nil {
+		return fmt.Errorf("certificate watcher was already initialized")
+	}
+
 	directory := ti.CertsDirectory
-	filesChanged, watcherCloser, err := watchDirectory(directory)
+	ti.updateCertificates(directory)
+
+	ti.watch = filewatch.New()
+	err := ti.watch.Add(directory, func() {
+		ti.updateCertificates(directory)
+	})
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("error adding directory to watch: %w", err)
 	}
 
-	ti.stopCertReload = make(chan struct{})
-	notify(filesChanged)
-
+	ti.stopWatch = make(chan struct{})
 	go func() {
-		defer watcherCloser.Close()
 		for {
 			select {
-			case <-filesChanged:
-				err := ti.updateCertificates(directory)
-				if err != nil {
-					go func() {
-						time.Sleep(retryInterval)
-						notify(filesChanged)
-					}()
-				}
-			case <-ti.stopCertReload:
+			case <-ti.stopWatch:
 				return
+			default:
```

Comment:
Is this required to satisfy the linter?

Reply:
no. It is to make the `select` non-blocking. It only checks if the `ti.stopWatch` channel is closed, and if not, it continues.